### PR TITLE
feat(training,#10211): holdout temporel 06/08 dans c1330 + le test DM est aveugle au signe

### PR DESCRIPTION
**Pull request by myia-ai-01**
Grain: DEEP/training — lane myia-ai-01:CoursIA — prev: MED/guard #10142

## Résumé

Tâche 2 de #10211 : la section 6 du notebook `c1330_xrp_dt_foldwise_research.ipynb` porte désormais les deux runs de **holdout temporel du 06/08**, lus dans leurs JSON. Les tâches 1 et 3 ont été livrées par #10215 (po-2025) ; cette PR complète la partition annoncée sur [#10211 (comment)](https://github.com/jsboige/CoursIA/issues/10211#issuecomment-5233737257).

**Pourquoi cette tâche est chez ai-01 et pas chez la lane qui tient le claim** : les deux JSON vivent sous `results/`, gitignoré (`.gitignore:51`), donc ils n'existent que sur la machine qui a fait tourner l'extension multi-seed le 06/08. Le critère exigeait des chiffres **issus d'une lecture du fichier**, pas d'une recopie : aucune autre lane ne pouvait le satisfaire sans recopier à la main ce que le critère interdit. Le claim de `myia-po-2025:CoursIA-2` n'est **pas** levé, et le crédit de l'analyse lui revient.

## Ce que la section établit — trois résultats, tous exécutés

Cellules 7, 8, 9 réellement exécutées, outputs réels committés (C.2).

**1. Le momentum naked bat le DT dans les DEUX fenêtres holdout.** La section 4 du notebook posait déjà le momentum comme « le vrai adversaire, pas BH » ; le holdout le confirme hors échantillon et plus nettement :

| Run | `train_end` | Obs | DT net | BH | momentum | edge vs mom | edge vs BH | Verdict |
|---|---|---:|---:|---:|---:|---:|---:|---|
| interne | 2025-06-30 | 294 | −1,8648 | −0,5924 | **−0,0342** | **−1,8306** | −1,2724 | `NO-BEATS` |
| frais | 2026-04-21 | 96 | +2,0985 | −1,8032 | **+1,6236** | **+0,4749** | +3,9017 | `INCONCLUSIVE` |

Presque tout l'« edge vs BH » du run frais (+3,90) est du momentum ; le gain marginal du Decision Transformer est **+0,47**. Et le comparateur `bh` est **dégénéré** dans cette fenêtre (BH = −1,8032) : battre un actif qui tombe en pouvant se mettre short n'est pas un résultat.

**2. `edge_sigma` change de signe selon la coupure** : −6,63 (interne) / +19,97 (frais). Son dénominateur ([`validate_xrp_dt_holdout.py:238`](MyIA.AI.Notebooks/QuantConnect/ML-Training-Pipeline/scripts/validate_xrp_dt_holdout.py)) est la dispersion **entre seeds entraînées sur les mêmes données** — il mesure la reproductibilité de la procédure, pas la significativité de l'edge. C'est le constat de #10211, reconduit ici avec les deux fenêtres côte à côte.

**3. Le test Diebold-Mariano tel qu'il est câblé est AVEUGLE AU SIGNE.** C'est le résultat neuf de cette PR, et il est **démontré par une cellule exécutée**, pas déduit d'une lecture de code :

```
serie      sharpe     dm_stat        p_value
----------------------------------------------
gagnante   +0.221   +0.757651   0.449257
perdante   -0.221   +0.757651   0.449257

dm_stat identiques : True
p_value identiques : True
```

Une série et son **exact opposé** reçoivent un `dm_stat` et un `p_value` identiques au bit. Cause mécanique : le site d'appel ([`validate_xrp_dt_holdout.py:132`](MyIA.AI.Notebooks/QuantConnect/ML-Training-Pipeline/scripts/validate_xrp_dt_holdout.py)) passe `-dt_n` / `-bh_g` avec `loss_fn='mse'`, et [`dm_test.py`](MyIA.AI.Notebooks/QuantConnect/ML-Training-Pipeline/scripts/dm_test.py) calcule `loss = errors ** 2` — le carré annule **la négation du site d'appel et le signe des rendements**. Le test compare `dt_net²` à `bh_gross²`, donc deux **volatilités réalisées**.

**Conséquence mesurée sur nos propres données** : sur le holdout interne, les seeds 42 et 99 sortent `p = 0,0421` et `p = 0,0182`, c'est-à-dire **significatifs à 5 % « en faveur du DT »** — alors que ces deux modèles font −1,62 et −1,81 de Sharpe contre −0,59 pour BH. Le test ne se contente pas d'être muet sur la rentabilité : il rend des verdicts significatifs **du mauvais côté**. Le `dm_p_median = 0,0847` qui a produit le `NO-BEATS` est la médiane de p-values dont deux auraient, prises seules, validé le troisième conjoint de §C pour une stratégie perdante.

## Portée — ce que cette PR ne fait pas

- **Elle ne corrige pas le validateur.** Le correctif (test sur le différentiel de rendement, ou une perte qui préserve le signe) est **hors périmètre** : il changerait les chiffres sans re-tourner les modèles, et les JSON ne persistent pas les séries de rendements, donc les p-values ne sont pas recalculables post hoc. Issue de suivi ouverte séparément (principe 3 : signaler, traiter en sujet dédié).
- **Elle n'invalide pas les `NO-BEATS` déjà rendus** — ils tombent sur les deux premiers conjoints de §C (`seeds > BH` et `edge ≥ 2σ`), pas sur le DM.
- **Elle ne réfute pas L4.** Le `BEATS` du registre reste vrai là où il a été mesuré (panel transversal @10 bps) ; il ne s'étend pas au holdout temporel. C'est un bornage de portée.

## Conception de la cellule de chargement

Pas de snapshot embarqué en fallback — contrairement à la section 0, qui en a un pour le sweep. Le critère du grain exige que les chiffres viennent du fichier : embarquer une copie rouvrirait la porte à la recopie manuelle. Sur une machine sans les JSON, la cellule imprime `[load] ABSENT` et **nomme la commande de régénération** (`python scripts/validate_xrp_dt_holdout.py --label <l> --coin XRP-USD`) — jamais un `except` muet (interdit harnais). La provenance est imprimée dans l'output : chemin, taille, `sha1` tronqué (`6e338a634bcf` / `30deedc9ca27`).

## Vérifications (relancées après le dernier commit)

```
notebook_tools validate         OK: 1   Warnings: 0   Errors: 0
check_c2_compliance --path      C.2 Compliance Check: 1/1 notebooks compliant — All clear!
strip_machine_paths --scan      0 notebook(s) carrying 0 leak line(s)
detect_papermill_path_leak      []
grep -cE "raise NotImplementedError|assert False|1/0"   -> 0
pre-commit (6 hooks)            tous Passed (gitleaks, probeAddresses, nuget-leak,
                                papermill-paths, md-rendering, H.3 null-exec)
```

`execution_count` : 1..9 sans trou, chaque cellule code porte ses outputs. Les cellules 1 à 6 ont été re-exécutées et restent identiques en substance — le sweep JSON `foldwise_20260806_165146.json` est toujours absent de cette machine, donc la section 0 retombe sur le même snapshot embarqué qu'avant (output inchangé). Les chemins papermill ont été normalisés au basename par [`scrub_papermill_paths.py`](scripts/notebook_tools/scrub_papermill_paths.py) — l'outil dédié, pas une édition manuelle.

Le PNG produit par la cellule 6 (`c1330_xrp_dt_foldwise_plots.png`) reste **non tracké**, comme avant cette PR : il n'est pas ajouté au dépôt.

## Note d'honnêteté sur un critère

Le critère de #10211 demandait la présence de `-1.865`. La valeur exacte du JSON est **`-1.8648`** (`aggregate.dt_net_sharpe_mean`), et c'est celle que le notebook imprime : j'avais écrit l'arrondi dans l'issue. Le critère est tenu en substance, avec une précision supérieure — pas avec la chaîne littérale. Les quatre autres (`-6.63`, `0.0847`, `+19.97`, `0.236`) apparaissent tels quels dans l'output de la cellule 8.

See #10211 (Tâche 2 ; Tâches 1 et 3 livrées par #10215), #1454.

### CODE REVIEW

**Review by jsboige — COMMENTED** [review]
[Hermes] COMMENT — c1330 holdout temporel 06/08

Vérifié cellule par cellule sur e4a4135 :

**Outputs authentiques** — timestamps cohérents (2026-08-09T20:52:xx), papermill durations plausibles (cell 9 = 4.96s de calcul réel), SHA1 des JSON imprimés dans la sortie. Pas de snapshot embarqué en fallback : la cellule 7 nomme explicitement le script régénérateur si les JSON sont absents.

**Le finding DM est mathématiquement correct.** Sous `loss_fn='mse'`, `dm_test.py` calcule `loss = errors**2`, donc `-gagnante**2 == perdante**2` bit-à-bit. La démonstration de la cellule 9 (dm_stat identiques pour Sharpe +0.22 vs −0.22) confirme que le troisième conjoint de §C mesure la volatilité réalisée, pas la rentabilité. C'est un résultat méthodologique important, correctement borné (« n'invalide pas les NO-BEATS existants, mais rend un futur BEATS non concluant »).

**Le bornage de portée est honnête** — NO-BEATS interne (0/5 seeds > BH), INCONCLUSIVE frais (dm_p=0.236 > 0.05), et le L4 BEATS reste valide uniquement pour le panel transversal @10bps. Le momentum naked bat le DT dans les deux fenêtres, et l'edge_sigma qui passe de −6σ à +20σ entre coupures est justement interprété comme une mesure de fenêtre, pas un edge.

**Une observation** : le changement Python 3.11→3.10 dans les metadata kernel est cohérent avec l'exécution sur ai-01 (et non po-2024), mais cela signifie que ce notebook n'est plus reproductible sur le runner GPU standard sans downgrade. Si c'est intentional et permanent, OK ; si un futur run revient sur po-2024, la version va re-flipper.

Bonne analyse, verdicts honnêtes, pas de security issue.
